src/System.IO.Pipelines/BufferSegment.cs

```diff
@@ -85,11 +85,11 @@ public BufferSegment(OwnedMemory<byte> buffer, int start, int end)
 
         public void Dispose()
         {
-            Debug.Assert(_buffer.ReferenceCount >= 1);
+            Debug.Assert(!_buffer.HasZeroReferences);
```

Comment:
I named it DisposableReference to highlight to users that they need to dispose it.